src/pages/smart-contracts/clarity-values-guide.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Clarity Values Guide
+description: Learn how to deal with Clarity Values in JavaScript.
+tags:
+  - tutorial
+---
+
+## Introduction
+The Clarity language makes use of a strong static [type system](https://docs.blockstack.org/references/language-clarity#clarity-type-system). This simply means that every function defined in Clarity expects arguments of specific types, and that a failure to provide arguments of the specified types will result in your code failing to compile, or your `contract-call` transaction failing prior to execution.
+
+In order to build web applications that interact with Clarity contracts, we must include type information as a part of our function calls. The [@stacks/transactions](https://github.com/blockstack/stacks.js/tree/master/packages/transactions) library makes this easy, as we will demonstrate below.
+
+## Clarity Types
+The Clarity type system includes the following types:
+
+- `(tuple (key-name-0 key-type-0) (key-name-1 key-type-1) ...)` - a typed tuple with named fields.
```

Comment:
This type corresponds to the ECMAScript [Record](https://github.com/tc39/proposal-record-tuple), an immutable mapping from keys to values. The `tuple` type in Clarity is a [misnomer](https://github.com/blockstack/stacks-blockchain/issues/1520), likely to confuse JS devs as TC39 will also introduce a `tuple` datatype in ECMAScript that is notably different than the `tuple` in Clarity.